src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -859,8 +859,12 @@ protected virtual async Task<bool> TryLoadFromCacheAsync(JsonOperationContext co
                     return;
 
                 var refreshTask = RefreshIfNeeded(chosenNode, response);
+                
                 command.StatusCode = response.StatusCode;
 
+                if (response.IsSuccessStatusCode || command.StatusCode == HttpStatusCode.NotModified)
+                    command.Etag = response.GetEtagHeader() ?? cachedChangeVector;
+
```

Comment:
Why is this needed?

Reply:
to handle the `NotModified` on the orchestrator level